a10_octavia/controller/worker/tasks/a10_network_tasks.py

```diff
@@ -672,90 +675,123 @@ class HandleVRIDFloatingIP(BaseNetworkTask):
     """Handle VRID floating IP configurations for loadbalancer resourse"""
 
     def __init__(self, *arg, **kwargs):
-        self.fip_port = None
+        self.added_fip_ports = []
         super(HandleVRIDFloatingIP, self).__init__(*arg, **kwargs)
 
     @axapi_client_decorator
     def execute(self, vthunder, lb_resource, vrid_list, subnet):
-        vrid = None
-
-        device_vrid_ip = None
-        vrid_floating_ip_list = []
-
-        for vr in vrid_list:
-            if vr.subnet_id == subnet.id:
-                vrid = vr
-                device_vrid_ip = vrid.vrid_floating_ip
-            else:
-                vrid_floating_ip_list.append(vr.vrid_floating_ip)
-        if vrid:
-            vrid_list.remove(vrid)
+        """
 
+        :param vthunder:
+        :param lb_resource: Can accept LB or member
+        :param vrid_list: VRID object list for LB resource's project.
+        :param subnet: subnet of the resource in question, will be helpful if there is no
+        VRID object present for the provided subnet then is should create new VRID
+        floating IP instead of updating existing(delete + create -> update)
+        :return: return the update list of VRID object, If empty the need to remove all VRID
+        objects from DB else need update existing ones.
+        """
+        vrid_floating_ips = []
+        update_vrid_flag = False
+        vrid_value = CONF.a10_global.vrid
         conf_floating_ip = a10_utils.get_vrid_floating_ip_for_project(
             lb_resource.project_id)
+        prev_vrid_value = copy.deepcopy(
+            vrid_list[0].vrid) if vrid_list else None
+
         if conf_floating_ip:
-            subnet_ip, subnet_mask = a10_utils.get_net_info_from_cidr(
-                subnet.cidr)
+            for vr in vrid_list:
+                if vr.subnet_id == subnet.id:
```

Comment:
What's up with this section? I see we check the ids and then break if they are equal. But I don't see anything actually change

Reply:
Here we iterate through VRID_LIST from db to check if the subnet for which operation is undergoing already has VRID FIP associated, If not we add object new object to the list through else which indicates creation operation other wise we do update existing vrid operation.

Reply:
```
      if conf_floating_ip:
            for vr in vrid_list:
                if vr.subnet_id == subnet.id:
                       break
      else:
                vrid_list.append(
                    data_models.VRID(
                        id=uuidutils.generate_uuid(),
                        vrid=vrid_value,
                        project_id=lb_resource.project_id,
                        vrid_port_id=None,
                        vrid_floating_ip=None,
                        subnet_id=subnet.id))
            if conf_floating_ip.lower() == 'dhcp':
```

Reply:
That's not what the code shows here